docs/features/deconstruction.md

```diff
@@ -129,65 +130,47 @@ evaluate side-effect on the left-hand-side variables
 evaluate Deconstruct passing the references directly in
 ```
 
-In the case where the expression on the right is a tuple, the evaluation order becomes:
-```
-evaluate side-effect on the left-hand-side variables
-evaluate the right-hand-side and do a tuple conversion (using a fake tuple representing the types in the left-hand-side)
-assign element-wise from the right to the left
-```
-
 #### Resolution of the Deconstruct method
 
 The resolution is equivalent to typing `rhs.Deconstruct(out var x1, out var x2, ...);` with the appropriate number of parameters to deconstruct into.
 It is based on normal overload resolution.
-This implies that `rhs` cannot be dynamic.
+This implies that `rhs` cannot be dynamic and that none of the parameters of the `Deconstruct` method can be type arguments. A `Deconstruct<T>(out T x1, out T x2)` method will not be found.
 Also, the `Deconstruct` method must be an instance method or an extension (but not a static method).
 
-#### Tuple Deconstruction
 
-Note that tuples (`System.ValueTuple`) don't need to invoke Deconstruct.
-`System.Tuple` are not recognized as tuples, and so will rely on Deconstruct (which will be provided for up to 3 nestings deep, that is 21 elements)
+###Deconstruction-declaration (deconstruction into new variables):
+
+Deconstruction-declarations can be thought of as two steps: (1) declaring new locals, and (2) applying a deconstruction-assignment into those locals.
+The syntax for deconstruction is more involved and multiple forms are allowed. The simplest case is `(int x, string y)`. Variants include nested declarations like `(int x, (string y, long z))` (which declares 3 locals) and implicitly typed declarations like `(var x, var y)`. The latter can also be written using the shorthand `var (x, y)`.
 
+As in the case of deconstruction-declarations, tuple expressions on the right-hand-side have their type inferred from the left-hand-side. With deconstruction-declaration, this is also the case, except that any type is `var` in the left-hand-side, then the natural type of the element in the right-hand-side is used.
+For example, in `(string x, byte y, var z) = (null, 1, 2);`, `null` has type `string`, the literal `1` has type `byte` (inferred from `y`) and the literal `2` has type `int` (its natural type).
 
-###Deconstruction-declaration (deconstruction into new variables):
+###Grammar changes
```

Comment:
You should probably mention, somewhere, that when the syntax `type ( variable_designation, ... )` is used, *type* must be `var`.